lib/puppet/type/archive.rb

```diff
@@ -122,9 +122,9 @@ def should_to_s(value)
   end
 
   newparam(:source) do
-    desc 'archive file source, supports http|https|ftp|file|s3 uri.'
+    desc 'archive file source, supports puppet|http|https|ftp|file|s3 uri.'
     validate do |value|
-      unless value =~ URI.regexp(%w(http https ftp file s3)) || Puppet::Util.absolute_path?(value)
+      unless value =~ URI.regexp(%w(puppet http https ftp file s3)) || Puppet::Util.absolute_path?(value)
         raise ArgumentError, "invalid source url: #{value}"
       end
```

Comment:
Follow up with chat with @alexjfisher, it might be worthwhile to mention in README, puppet:/// source will use ruby provider regardless of provider setting for now. But not a blocker for merging.